<launch>
	<!-- Base Parameters -->
	<arg name="project_directory" default="/Titan/code/robohike_ws/src/topo_loc" />
	<arg name="map_path" default="/Rocket_ssd/dataset/data_topo_loc/ucl_campus/out/out_map/" />

	<param name="use_sim_time" value="true" />

	<!-- Localization -->
	<node name="ros_loc_pipeline" pkg="topo_loc" type="ros_loc_pipeline.py" output="screen"
		args="--dataset_path $(arg map_path)
					--device=cuda --vpr_method cosplace --vpr_backbone=ResNet18 --vpr_descriptors_dimension=512 
					--img_matcher master --pose_solver pnp
					--config_pose_solver $(arg project_directory)/python/config/dataset/ucl_campus.yaml --no_viz
					--ros_rgb_img_type compressed" >
		<remap from="/color/image" to="/zed2/zed_node/left/image_rect_color/compressed" />
		<remap from="/depth/image" to="/zed2/zed_node/depth/depth_registered" />
		<remap from="/color/camera_info" to="/zed2/zed_node/left/camera_info" />					
	</node>

	<node name="ros_pose_fusion" pkg="topo_loc" type="ros_pose_fusion.py" output="screen">
		<remap from="/local/odometry" to="/state_estimator/odometry" />
		<remap from="/global/odometry" to="/vloc/odometry" />
		<param name="frame_id_map" value="vloc_map" />
	</node>

  <!-- Static TF -->
  <!-- Option 1: x y z yaw pitch roll (radians) frame_id child_frame_id period_in_ms -->
  <!-- Option 2: x y z qx qy qz qw frame_id child_frame_id period_in_ms -->
  <node pkg="tf" type="static_transform_publisher" name="vloc_trans_publisher" args="0 0 0 0 0 0 map vloc_map 10" />

</launch>
